shared/src/commonMain/kotlin/co/nimblehq/blisskmmic/data/network/target/UserTargetType.kt

```diff
@@ -6,9 +6,10 @@ import io.ktor.http.*
 import kotlinx.serialization.SerialName
 import kotlinx.serialization.Serializable
 
-sealed class UserTargetType: TargetType
+sealed class UserTargetType<T>: TargetType<T>
 
-class LoginTargetType(email: String, password: String): UserTargetType() {
+class LoginTargetType(email: String, password: String):
+    UserTargetType<LoginTargetType.LoginInput>() {
```

Comment:
It quite duplicates when we keep defining sealed class.
```
sealed class SurveyTargetType<T>: TargetType<T>
sealed class UserTargetType<T>: TargetType<T>
```

Can we do like this?
```suggestion
class LoginTargetType(email: String, password: String):
    TargetType<LoginTargetType.LoginInput>() {
```